libs/feature/editor/src/lib/components/record-form/form-field/form-field-contacts-for-resource/form-field-contacts-for-resource.component.ts

```diff
@@ -0,0 +1,220 @@
+import { CommonModule } from '@angular/common'
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  Input,
+  OnChanges,
+  OnInit,
+  SimpleChanges,
+  Type,
+} from '@angular/core'
+import { FormControl } from '@angular/forms'
+import {
+  AutocompleteComponent,
+  DropdownSelectorComponent,
+  UiInputsModule,
+} from '@geonetwork-ui/ui/inputs'
+import { UiWidgetsModule } from '@geonetwork-ui/ui/widgets'
+import {
+  Individual,
+  Organization,
+  RoleLabels,
+} from '@geonetwork-ui/common/domain/model/record'
+import { TranslateModule } from '@ngx-translate/core'
+import { debounceTime, distinctUntilChanged, Observable, switchMap } from 'rxjs'
+import { UserModel } from '@geonetwork-ui/common/domain/model/user'
+import { PlatformServiceInterface } from '@geonetwork-ui/common/domain/platform.service.interface'
+import { OrganizationsServiceInterface } from '@geonetwork-ui/common/domain/organizations.service.interface'
+import { ContactCardComponent } from '../../../contact-card/contact-card.component'
+import {
+  DynamicElement,
+  SortableListComponent,
+} from '@geonetwork-ui/ui/elements'
+
+@Component({
+  selector: 'gn-ui-form-field-contacts-for-resource',
+  templateUrl: './form-field-contacts-for-resource.component.html',
+  styleUrls: ['./form-field-contacts-for-resource.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  standalone: true,
+  imports: [
+    DropdownSelectorComponent,
+    UiInputsModule,
+    CommonModule,
+    UiWidgetsModule,
+    AutocompleteComponent,
+    TranslateModule,
+    ContactCardComponent,
+    SortableListComponent,
+  ],
+})
+export class FormFieldContactsForResourceComponent
+  implements OnInit, OnChanges
+{
+  @Input() control: FormControl<Individual[]>
+
+  allUsers$: Observable<UserModel[]>
+
+  rolesToPick: string[] = [
+    'resource_provider',
+    'custodian',
+    'owner',
+    'point_of_contact',
+    'author',
+  ]
+
+  rolesToDisplay = []
+
+  allOrganizations: Map<string, Organization> = new Map()
+
+  addOptions: Array<{ buttonLabel: string; eventName: string }> = []
+
+  constructor(
+    private platformServiceInterface: PlatformServiceInterface,
+    private organizationsServiceInterface: OrganizationsServiceInterface,
+    private changeDetectorRef: ChangeDetectorRef
+  ) {
+    this.allUsers$ = this.platformServiceInterface.getUsers()
+  }
+
+  ngOnInit(): void {
+    this.organizationsServiceInterface.organisations$.subscribe(
+      (organizations) => {
+        this.allOrganizations = new Map<string, Organization>(
+          organizations.map((organization) => [organization.name, organization])
+        )
+        this.changeDetectorRef.markForCheck()
+      }
+    )
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
```

Comment:
See my comment below about using getters, that would mean that you wouldn't have to rely on ngOnChanges anymore